src/dp_vnf.c

```diff
@@ -94,16 +102,17 @@ int dp_del_vnf(const uint8_t ul_addr6[DP_VNF_IPV6_ADDR_SIZE])
 
 static __rte_always_inline bool dp_vnf_match(const struct dp_vnf *vnf, enum dp_vnf_type type,
 											 uint16_t port_id, uint32_t vni,
-											 uint32_t prefix_ip, uint8_t prefix_len)
+											 struct dp_ip_address *pfx_ip, uint8_t prefix_len)
 {
 	return (port_id == DP_VNF_MATCH_ALL_PORT_IDS || vnf->port_id == port_id)
 		&& vnf->vni == vni
-		&& vnf->alias_pfx.ip == prefix_ip
+		&& ((vnf->alias_pfx.ol.ip_type == RTE_ETHER_TYPE_IPV4 && vnf->alias_pfx.ol.ipv4 == pfx_ip->ipv4)
+			|| (vnf->alias_pfx.ol.ip_type == RTE_ETHER_TYPE_IPV6 && rte_rib6_is_equal(vnf->alias_pfx.ol.ipv6, pfx_ip->ipv6)))
 		&& vnf->alias_pfx.length == prefix_len
 		&& vnf->type == type;
```

Comment:
This comparison seems to not check `pfx_ip->ip_type`, not sure if intentional. I guess if the caller makes sure to only have the right type, then it's an optimization, but would need to check.

Reply:
Adding it would make sense, we may have the case where the lower bytes of the ipv6 address matches to ipv4 comparison :)